framework/src/controller/application.js

```diff
@@ -224,6 +224,10 @@ class Application {
 		);
 		assert(Transaction, 'Transaction implementation is required');
 
+		if (options.matcher) {
+			Transaction.matcher = options.matcher;
+		}
+
 		const transactions = this.getTransactions();
 		transactions[transactionType] = Object.freeze(Transaction);
```

Comment:
We are freezing couple of stuff right before we start the application. I would suggest to move this logic to that part. 

https://github.com/LiskHQ/lisk-sdk/blob/bb52221265d0f053109eb864b6593f9b389f6a34/framework/src/controller/application.js#L282-L284

Reply:
I agree that it would concentrate similar actions in one place but it would add an unnecessary step imo.